dll/win32/uxtheme/nonclient.c

```diff
@@ -1121,22 +1233,88 @@ HRESULT WINAPI DrawNCPreview(HDC hDC,
+    DrawWindowForNCPreview(hDC, &context, rcAdjPreview.left, rcAdjPreview.top, rcAdjPreview.right - 17, rcAdjPreview.bottom - 20, TRUE, NULL);
+
+    /* Draw active preview window */
+    context.Active = TRUE;
+    SetWindowTextW(hwndDummy, L"Active Window");
+
+    DWORD textDrawFlags = DT_NOPREFIX | DT_SINGLELINE | DT_WORDBREAK;
+    RECT rcWindowClient;
+    DrawWindowForNCPreview(hDC, &context, rcAdjPreview.left + 10, rcAdjPreview.top + 22, rcAdjPreview.right, rcAdjPreview.bottom, TRUE, &rcWindowClient);
+    SetTextColor(hDC, GetThemeSysColor(context.theme, COLOR_WINDOWTEXT));
+    LOGFONTW lfText;
+    HFONT textFont = NULL;
+    if (SUCCEEDED(GetThemeSysFont(context.theme, TMT_MSGBOXFONT, &lfText)))
+        textFont = CreateFontIndirectW(&lfText);
+
+    if (textFont)
+        SelectFont(hDC, textFont);
+
+    DrawThemeText(context.theme, hDC, WP_DIALOG, 0, L"Window Text", -1, DT_LEFT | DT_TOP | textDrawFlags, 0, &rcWindowClient);
+
+    /* Draw preview dialog window */
+    SetWindowTextW(hwndDummy, L"Message Box");
+    DWORD dwStyleNew = WS_VISIBLE | WS_OVERLAPPED | WS_CAPTION | WS_SYSMENU | WS_DLGFRAME;
+    SetWindowLongPtr(hwndDummy, GWL_STYLE, dwStyleNew);
+    DWORD dwExStyleNew = WS_EX_DLGMODALFRAME;
+    SetWindowLongPtr(hwndDummy, GWL_EXSTYLE, dwExStyleNew);
+
+    if (!GetWindowInfo(hwndDummy, &context.wi))
+        return E_FAIL;
+
+    context.wi.dwStyle = dwStyleNew;
+    context.wi.dwExStyle = dwExStyleNew;
+
+    INT msgBoxHalfWidth = 75;
+    INT msgBoxHCenter = rcAdjPreview.left + (previewWidth / 2);
+    INT msgBoxVCenter = rcAdjPreview.top + (previewHeight / 2);
+    if (previewHeight % 2 != 0)
+        msgBoxVCenter++;
+
+    DrawWindowForNCPreview(hDC, &context, msgBoxHCenter - msgBoxHalfWidth, msgBoxVCenter - 29, msgBoxHCenter + msgBoxHalfWidth, msgBoxVCenter + 71, FALSE, &rcWindowClient);
+    DrawThemeBackground(context.theme, hDC, WP_DIALOG, 0, &rcWindowClient, NULL);
+
+    /* Draw preview dialog button */
+    HTHEME hBtnTheme = OpenThemeDataFromFile(hThemeFile, hwndDummy, L"BUTTON", OTD_NONCLIENT);
+    if (hBtnTheme)
+    {
+        INT btnCenterH = rcWindowClient.left + ((rcWindowClient.right - rcWindowClient.left) / 2);
+        INT btnCenterV = rcWindowClient.top + ((rcWindowClient.bottom - rcWindowClient.top) / 2);
+        RECT rcBtn = { btnCenterH - 40, btnCenterV - 15, btnCenterH + 40, btnCenterV + 15};
+        int btnPart = BP_PUSHBUTTON;
+        int btnState = PBS_DEFAULTED;
+        DrawThemeBackground(hBtnTheme, hDC, btnPart, btnState, &rcBtn, NULL);
+        MARGINS btnContentMargins;
+        if (GetThemeMargins(hBtnTheme, hDC, btnPart, btnState, TMT_CONTENTMARGINS, NULL, &btnContentMargins) == S_OK)
+        {
+            rcBtn.left += btnContentMargins.cxLeftWidth;
+            rcBtn.top += btnContentMargins.cyTopHeight;
+            rcBtn.right -= btnContentMargins.cxRightWidth;
+            rcBtn.bottom -= btnContentMargins.cyBottomHeight;
+        }
+
+        LPCWSTR btnText = L"OK";
```

Comment:
Could be placed in the function call directly? Unless you open the possibility for this string to be localized later?

Reply:
It...should be localized, should it not?

Reply:
It would be better to localize later, one step at a time (tm)

Reply:
@binarymaster Right...so in that case, what should I do for right now?

Reply:
Leave it for now seems fine.
Let's address bugs and issues that should not be merged, improvements should not be piled on until this PR starts to rot again.

Reply:
Noted. I'll hold off on any further additions/improvements/etc. until then :+1:

Reply:
Yes it should, ultimately. See the very first comment from jeffbox at the top of this PR.